### PR TITLE
pkg/p2p: fix ErrPeerMessageDataLost after ErrPeerMessageTopicCongested is triggered

### LINKED ISSUE #5880: executor exits unexpected during job master failover
### What did you do?

Use a mock fake master to simulate job master error, setup one server master and one executor, create a fake job

```diff
➜  git diff
diff --git a/engine/framework/fake/fake_master.go b/engine/framework/fake/fake_master.go
index 565772471..c37fd2351 100644
--- a/engine/framework/fake/fake_master.go
+++ b/engine/framework/fake/fake_master.go
@@ -381,7 +381,7 @@ func (m *Master) tickedCheckStatus(ctx context.Context) error {
                return m.Exit(ctx, m.Status(), nil)
        }

-       return nil
+       return errors.New("injected fake error!!!!")
 }
```

### What did you expect to see?

_No response_

### What did you see instead?

The executor node exits after about 1 minute.

```go
[2022/06/15 12:17:31.184 +08:00] [INFO] [task_runner.go:249] ["Task Closed"] [id=944cb883-1bb0-4f86-892b-eee910179b7c] [] [runtime-task-count=1]
[2022/06/15 12:17:31.184 +08:00] [WARN] [client.go:168] ["peer message client detected error, restarting"] [error="rpc error: code = Aborted desc = CDC capture closing"] [errorVerbose="rpc error: code = Aborted desc = CDC capture closing\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.Trace\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/juju_adaptor.go:15\ngithub.com/pingcap/tiflow/pkg/p2p.(*MessageClient).runRx\n\tgithub.com/pingcap/tiflow/pkg/p2p/client.go:349\ngithub.com/pingcap/tiflow/pkg/p2p.(*MessageClient).run.func2\n\tgithub.com/pingcap/tiflow/pkg/p2p/client.go:205\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20220513210516-0976fa681c29/errgroup/errgroup.go:74\nruntime.goexit\n\truntime/asm_amd64.s:1571"]
[2022/06/15 12:17:31.186 +08:00] [ERROR] [main.go:87] ["run executor with error"] [error="[CDC:ErrWorkerPoolHandleCancelled]workerpool handle is cancelled"] [errorVerbose="[CDC:ErrWorkerPoolHandleCancelled]workerpool handle is cancelled\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/normalize.go:164\ngithub.com/pingcap/tiflow/pkg/workerpool.(*defaultEventHandle).AddEvent\n\tgithub.com/pingcap/tiflow/pkg/workerpool/pool_impl.go:132\ngithub.com/pingcap/tiflow/pkg/p2p.(*MessageServer).handleMessage\n\tgithub.com/pingcap/tiflow/pkg/p2p/server.go:788\ngithub.com/pingcap/tiflow/pkg/p2p.(*MessageServer).handlePendingMessages\n\tgithub.com/pingcap/tiflow/pkg/p2p/server.go:518\ngithub.com/pingcap/tiflow/pkg/p2p.(*MessageServer).registerHandler\n\tgithub.com/pingcap/tiflow/pkg/p2p/server.go:504\ngithub.com/pingcap/tiflow/pkg/p2p.(*MessageServer).run\n\tgithub.com/pingcap/tiflow/pkg/p2p/server.go:231\ngithub.com/pingcap/tiflow/pkg/p2p.(*MessageServer).Run.func2\n\tgithub.com/pingcap/tiflow/pkg/p2p/server.go:194\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20220513210516-0976fa681c29/errgroup/errgroup.go:74\nruntime.goexit\n\truntime/asm_amd64.s:1571"]
```

detail logs:
[executor1.log](https://github.com/pingcap/tiflow/files/8905656/executor1.log)
 

### Versions of the cluster

Dataflow Engine version (run `df-master -V` or `df-executor -V`):

master@https://github.com/pingcap/tiflow/commit/44ee050b64f91d9ba5cb0e5f08758152b31066df

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve?

Issue Number: close #5880 

### What is changed and how it works?
- Drops p2p messages without a corresponding valid peer.

#### Tests 

 - Unit test
 - Integration test

##### Will it cause performance regression or break compatibility?
No.

##### Do you need to update user documentation, design documentation or monitoring documentation?
No.

### Release note 

```release-note
None.
```

### CODE REVIEW

File: pkg/p2p/server.go [hunk 1]
```diff
@@ -408,7 +422,7 @@ func (m *MessageServer) AddHandler(
 				zap.Int64("lastAck", lastAck))
 			return nil
 		}
-
+		log.Info("sequence", zap.Int64("seq", entry.Sequence))
```
Comment (amyangfei) [inline pkg/p2p/server.go:425]: Will this log be useful if the log level is not `debug`.

Comment (liuzix) [reply pkg/p2p/server.go:425]: removed

**Review by amyangfei — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Pull request merged**